buildenv/jenkins/jobs/infrastructure/copyrightCheck

```diff
@@ -36,386 +39,405 @@ def ignorefile = "${env.WORKSPACE}/.copyrightignore"
-                                                foundIBMCopyright < foundCopyright) {
-                                                    //The IBM Copyright is not after the existing copyright in the file
-                                                    errorStr = "IBM Copyright is not after the existing copyright"
-                                                    FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                                errorStr = "IBM Copyright (basic) is missing from the file"
+                                                FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                                checkCopyrightDate = false
                                             }
-                                        } else {
-                                            errorStr = "IBM Copyright (basic) is missing from the file"
-                                            FAIL = addError(BAD_FILES, "${file}", errorStr)
-                                            checkCopyrightDate = false
                                         }
-                                    }
-                                } // !foundNonIBMCopyright
-                            } // if closed
-                        } else {
-                            // must be an eclipse repository, so just 
-                            // check that the date is valid
-                            checkCopyrightDate = true
-                            println "Checking file: '${file}'"
-                        }
-                            
-                         // We need to check that the copyright update year is the current year 
-                        if (checkCopyrightDate) {
-                            RESULT = sh (
-                                script: "grep -qE ${REGEX} '${file}'",
-                                returnStatus: true)
-                            if(RESULT != 0) {
-                                errorStr = "IBM Copyright date appears to be incorrect"
-                                FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                    } // !foundNonIBMCopyright
+                                } // if closed
                             } else {
-                                echo "Copyright date in file: appears to be correct"
+                                // must be an eclipse repository, so just
+                                // check that the date is valid
+                                checkCopyrightDate = true
+                                println "Checking file: '${file}'"
+                            }
+
+                             // We need to check that the copyright update year is the current year
+                            if (checkCopyrightDate) {
+                                RESULT = sh (
+                                    script: "set +x; echo '${theFile.substring(0,2000)}' | grep -qE ${REGEX}",
```

Comment:
Does this imply the copyright must be within the first 2000 characters or lines of the file? Does this hold true for all the files? I think there are some in extensions that don't appear at the very top.

Reply:
It is sufficient. However, since the file content is read, I reformatted the code to use String.find() function instead of grep to search for a groovy pattern.